### ISSUE TITLE
multiline-comment-style should ignore the comments at the start of a userscript if greasemonkey is true

### ISSUE THREAD

**Issue by revolter** [eslint/eslint#10673]
**Tell us about your environment**

* **ESLint Version:** 5.2.0
* **Node Version:** v8.9.4
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "browser": true,
        "es6": true,
        "greasemonkey": true
    },
    "extends": "eslint:recommended",
    "rules": {
        "multiline-comment-style": [
            "error",
            "bare-block"
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// ==UserScript==
// @name         Main
// @version      1
// ==/UserScript==

(function() {
    "use strict";

    1 + 1;
})();
```

```bash
eslint scripts/**/*
```

**What did you expect to happen?**

Not throw an error.

**What actually happened? Please include the actual, raw output from ESLint.**

```
.../scripts/Main/Main.js
  1:1  error  Expected a block comment instead of consecutive line comments  multiline-comment-style

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

**Comment by revolter** [comment]
This is still an issue.

**Comment by platinumazure** [comment]
I'm mildly opposed to adding this to core, as we can't have write a bunch of environment-specific exceptions into the core rules-- they would become unmaintainable very quickly.

I recommend creating a custom rule for this case (and, potentially, publishing it in a plugin). You could use [eslint-rule-composer](https://github.com/not-an-aardvark/eslint-rule-composer) to create a rule based on the existing core rule, and filter out reports of comments at the top of a file if the `greasemonkey` env is enabled.

**Comment by revolter** [comment]
It's not an exception, is the standard way of defining a GreaseMonkey script. So, knowing that ESLint already officially [supports GreaseMonkey](https://eslint.org/docs/user-guide/configuring#specifying-environments), this is definitely a bug. You can't have a GreaseMonkey script starting with:

```js

/*
 ==UserScript==
 @name         Main
 @version      1
 ==/UserScript==
*/

(function() {
    "use strict";

    1 + 1;
})();
```
